ckanext/datastore/backend/postgres.py

```diff
@@ -2537,14 +2589,17 @@ def resource_fields(self, id: str) -> dict[str, Any]:
                                            'uniquekey': row.uniquekey,
                                            'foreignkeys': row.foreignkeys}
 
-                # (canada fork only): foreign keys
-                # TODO: upstream contrib!!
-                #FIXME: fix later, multiple column foreign keys and ordered dict/lists?
-                if row.foreignkeys:
-                    colinfo['foreign_key_maps'] = _get_foreign_constraints(engine, id, column_name=colname)
-
                 schemainfo[colname] = colinfo
 
+            # (canada fork only): foreign keys
+            # TODO: upstream contrib!!
+            # constraint info requires higher perms
```

Comment:
we could instead find out what permissions are required and grant them to the R/O user in the set permissions sql that ckan generates for the datastore.

Reply:
will dig a bit to see what we would need for this, I imagine it is READ on pg_constraints

Reply:
@wardi yeah it is actually on `information_schema.referential_constraints`

Reply:
https://www.postgresql.org/docs/current/infoschema-referential-constraints.html

```
The view referential_constraints contains all referential (foreign key) constraints in the current database. Only those constraints are shown for which the current user has write access to the referencing table (by way of being the owner or having some privilege other than SELECT).
```